genestack_client/data_importer.py

```diff
@@ -490,15 +532,19 @@ def create_reference_genome(self,
         :return:
         """
         metainfo = metainfo or Metainfo()
-        name and metainfo.add_string(Metainfo.NAME, name)
-        organism and metainfo.add_string(BioMetaKeys.ORGANISM, organism)
-        strain and metainfo.add_string(BioMetaKeys.STRAIN, strain)
-        assembly and metainfo.add_string(BioMetaKeys.REFERENCE_GENOME_ASSEMBLY, assembly)
-        release and metainfo.add_string(BioMetaKeys.REFERENCE_GENOME_RELEASE, release)
-        annotation_url and metainfo.add_external_link(ANNOTATION_KEY, annotation_url, text='Annotations data link')
-        metainfo.add_string(metainfo.DESCRIPTION, description or '')
-        for seq_link in sequence_urls:
-            metainfo.add_external_link(SEQUENCE_KEY, seq_link, text='Sequence data link')
+        self.__add_to_metainfo(metainfo, Metainfo.NAME, name, StringValue, required=True)
+        self.__add_to_metainfo(metainfo, BioMetaKeys.ORGANISM, organism, StringValue)
+        self.__add_to_metainfo(metainfo, BioMetaKeys.STRAIN, strain, StringValue)
+        self.__add_to_metainfo(metainfo, BioMetaKeys.REFERENCE_GENOME_ASSEMBLY, assembly, StringValue)
+        self.__add_to_metainfo(metainfo, BioMetaKeys.REFERENCE_GENOME_RELEASE, release, StringValue)
+        self.__add_to_metainfo(metainfo, ANNOTATION_KEY, annotation_url, ExternalLink)
+        self.__add_to_metainfo(metainfo, metainfo.DESCRIPTION, description, StringValue)
+
+        if sequence_urls:
```

Comment:
> I can't see why we test only the first URL

Because if test fro first url is passed, we don't need to test other and cannot test other because this will provide an error.

I can move this test into  test function (if is_list ....) or wait a little and implement ListValue support